src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,79 @@
 package core.basesyntax;
 
-public class WorkWithFile {
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
 
+public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        File file = new File(fromFileName);
+
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
+            StringBuilder stringBuilder = new StringBuilder();
+            String value = bufferedReader.readLine();
+
+            while (value != null) {
+                stringBuilder.append(value).append(" ");
+                value = bufferedReader.readLine();
+            }
+
+            String supplyAndBuyToString = stringBuilder.toString();
+
+            String[] supplyAndBuy = supplyAndBuyToString.split(" ");
+
+            StringBuilder supplyBuilder = new StringBuilder();
+            StringBuilder buyBuilder = new StringBuilder();
+
+            for (String s : supplyAndBuy) {
+                if (s.charAt(0) == 's') {
+                    supplyBuilder.append(s).append(" ");
+                } else {
+                    buyBuilder.append(s).append(" ");
+                }
+            }
+            String stringSupply = supplyBuilder.toString();
+            String stringBuy = buyBuilder.toString();
+            String[] arraySupply = stringSupply.split("\\D+");
+            String[] arrayBuy = stringBuy.split("\\D+");
+            int[] supplyNumbers = new int[arraySupply.length - 1];
+            int[] buyNumbers = new int[arrayBuy.length - 1];
+            for (int i = 1; i < arraySupply.length; i++) {
+                int local = Integer.parseInt(arraySupply[i]);
+                supplyNumbers[i - 1] += local;
+            }
+            for (int i = 1; i < arrayBuy.length; i++) {
+                int local = Integer.parseInt(arrayBuy[i]);
+                buyNumbers[i - 1] += local;
+            }
+
+            int sumSupply = 0;
+            int sumBuy = 0;
+            int result;
+
+            for (int supplyNumber : supplyNumbers) {
+                sumSupply += supplyNumber;
+            }
+            for (int buyNumber : buyNumbers) {
+                sumBuy += buyNumber;
+            }
+
+            result = sumSupply - sumBuy;
+
+            String completedTextFromFileWriter = "supply," + sumSupply + System.lineSeparator()
+                    + "buy," + sumBuy + System.lineSeparator()
+                    + "result," + result;
+
+            File finalFile = new File(toFileName);
 
+            BufferedWriter writerToFinalFile = new BufferedWriter(new FileWriter(finalFile, true));
+            writerToFinalFile.write(completedTextFromFileWriter);
+            writerToFinalFile.close();
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file or can't write data to file", e);
+        }
```

Comment:
Divide program logic
Large methods is very difficult to read and understand. So methods should be short, between 5-15 lines. And it’s important for you to remember that one method should be responsible for one task.

For example: “write to file” or “create report” - are a separate operations. So we should create separate methods for them.


Use constants
If you have strange strings or numbers in the code, it’s better to declare them as constants. The name of the constant should display this object’s purpose.